Compiler/src/typeinfer.jl

```diff
@@ -1432,6 +1438,7 @@ function typeinf_ext_toplevel(methods::Vector{Any}, worlds::Vector{UInt}, trim_m
     inspected = IdSet{CodeInstance}()
     tocompile = Vector{CodeInstance}()
     codeinfos = []
+    bt = collect_dispatch_backtrace[] ? backtrace() : nothing
```

Comment:
This method is not actually relevant to dispatch, so these additional changes (just the first one) should not be included in this PR